internal/proxy/task_scheduler.go

```diff
@@ -170,12 +172,17 @@ func (queue *baseTaskQueue) Enqueue(t task) error {
 		return err
 	}
 
-	ts, err := queue.tsoAllocatorIns.AllocOne(t.TraceCtx())
-	if err != nil {
-		return err
+	var ts Timestamp
+	if t.CanSkipAllocTimestamp() {
+		physicalTs := time.Now().UnixMilli()
+		ts = tsoutil.ComposeTS(physicalTs, rand.Int63n(physicalTs))
+	} else {
+		ts, err = queue.tsoAllocatorIns.AllocOne(t.TraceCtx())
+		if err != nil {
+			return err
+		}
+		t.SetTs(ts)
```

Comment:
Why we don't `SetTs` for tasks which can skip allocating timestamp?

Reply:
This timestamp will be set to MVCCTS. In querynode, there is logic that if MVCCTS is 0, it will be assigned the value tsafe. If ts is set here, it may affect the query.

https://github.com/czs007/milvus/blob/b9d76f77d19810f8f7c8f760745bdb5c6495c61f/internal/querynodev2/delegator/delegator.go#L290-L292

Reply:
https://github.com/czs007/milvus/blob/b9d76f77d19810f8f7c8f760745bdb5c6495c61f/internal/querynodev2/delegator/delegator.go#L290-L292